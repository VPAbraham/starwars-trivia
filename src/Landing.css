section {
  color: #ffffff;
  display: flex;
  flex-direction: column;
  font-size: 2.0rem;
  align-items: center;
}

h2 {
  margin: 8px;
}

form {
  align-items: center;
  background-color: rgb(90, 104, 208, .5);
  /* opacity: .7; */
  display: flex;
  flex-direction: column;
  height: 60vh;
  width: 40%;
}

label {
  margin: 5px;
  text-align: center;
}

input,
select {
  background-color: rgb(90, 104, 208, .8);
  border: 1.5px solid white;
  font-size: 1.5rem;
  height: 6vh;
  margin: 10px;
  opacity: unset;
  padding: 5px;
  width: 90%;
}

button {
  background: rgb(90, 104, 208, .5);
  border: none;
  border-top: #ffffff 2px solid;
  border-bottom: #ffffff 2px solid;
  border-radius: 30px;
  color: #ffffff;
  font-size: 2rem;
  height: 6vh;
  width: 75%;
}

