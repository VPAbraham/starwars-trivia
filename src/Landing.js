import React, { Component } from 'react';



class Landing extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <section>
        <h1>StarWars</h1>
        <h2>Nerd Trove</h2>
        <form>
          <input type='text' name='userName' />
          <input type='text' name='movieQuote' />
          <select required>
            <option value='' selected>Select your strength in the force...</option>
            <option value='padawan'>Padawan</option>
            <option value='jedi knight'>Jedi Knight</option>
            <option value='jedi master'>Jedi Master</option>
          </select>
        </form>
      </section>
    )
  }
}



export default Landing;