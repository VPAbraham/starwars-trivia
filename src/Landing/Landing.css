section {
  color: #ffffff;
  display: flex;
  flex-direction: column;
  font-size: 2.0rem;
  align-items: center;
}

h2 {
  margin: 8px;
}

form {
  align-items: center;
  background-color: rgb(90, 104, 208, .5);
  /* opacity: .7; */
  display: flex;
  flex-direction: column;
  height: 65vh;
  width: 40%;
}

label {
  margin: 5px;
  text-align: center;
}

input,
select {
  background-color: rgb(90, 104, 208, .8);
  border: 1.5px solid white;
  font-size: 1.5rem;
  height: 6vh;
  margin: 10px;
  opacity: unset;
  padding: 5px;
  width: 90%;
}

Link,
button {
  background: rgb(90, 104, 208, .5);
  border: none;
  border-top: #ffffff 2px solid;
  border-bottom: #ffffff 2px solid;
  border-radius: 30px;
  color: #ffffff;
  font-size: 2rem;
  height: 6vh;
  width: 15vw;
}

.footer--user-info,
.footer--user-bar {
  background: #000000;
  bottom: 0;
  color: #ffffff;
  display: flex;
  justify-content: space-around;
  height: 10vh;
  margin: 0;
  padding: 1%;
  position: absolute;
  width: 100vw;
}

.footer--user-bar {
  z-index: 10;
}

.footer--user-info {
  bottom: 0;
}

.p--user-info {
  margin: 0;
}

