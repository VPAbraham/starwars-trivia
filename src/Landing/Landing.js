import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Error from '../Error/Error';
import StarWarsLogo from '../images/StarWarsLogo.png';
import './Landing.css';


class Landing extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      userQuote: '',
      skillLevel: '',
      formComplete: false,
      error: '',
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
    this.checkFormFields();
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.updateUserInfo(this.state); 
  }

  checkFormFields = () => {
    const keys = Object.keys(this.state);
    const userKeys = keys.filter(key => key.includes('user'));    
    const empty = userKeys.filter(key => this.state[key] === '');
    console.log(empty)
    return empty.length === 0 ? this.setState({ formComplete: true}) : this.setState({ formComplete: false});
  }

  handleError = (event) => {
    event.preventDefault()
    this.setState({ error: 'All Fields Required Padawan. You Must Focus!' });
  }

  clearError = (event) => {
    event.preventDefault();
    this.setState({ error: '' })
  }

  render() {
    return (
      <section className='section section--landing-form'>
        {/* <h1>StarWars</h1> */}
        <img className='img img--logo-main' src={StarWarsLogo} alt='star wars'/>
        <h2>Nerd Trove</h2>
        <form>
          <label htmlFor='userName'>Enter your name Jedi</label>
          <input type='text' name='userName' onChange={this.handleChange} />
          <label htmlFor='movieQuote'>Enter your favorite Star Wars quote</label>
          <input type='text' name='userQuote' onChange={this.handleChange}/>
          <label htmlFor='skillLevel'>What is your strength in the force?</label>
          <select name='skillLevel'required onChange={this.handleChange}>
            <option value=''>Select your strength in the force...</option>
            <option value='padawan'>Padawan</option>
            <option value='jedi knight'>Jedi Knight</option>
            <option value='jedi master'>Jedi Master</option>
          </select>
          {this.state.formComplete &&
            <button onClick={event => this.handleSubmit(event)}>
              <Link className='buttonLink' to='/movies'>
                Login
              </Link>
            </button>}
          {!this.state.formComplete &&
            <button onClick={event => this.handleError(event)}>
                Login</button>}
        </form>
        <footer className='footer footer--user-bar'>
        <p className='p p--user-info'>User Information Bar</p>
        </footer>
        {this.state.error && <Error msg={this.state.error} clearError={this.clearError}/>}
      </section>
    )
  }
}



export default Landing;