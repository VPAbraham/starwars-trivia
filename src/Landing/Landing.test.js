import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';
import { exportAllDeclaration } from '@babel/types';

describe('Landing', () => {
    const updateUserInfo = jest.fn();
    it('should match the snapshot', () => {
      const wrapper = shallow(
        <Landing updateUserInfo={updateUserInfo} />
      )
      expect(wrapper).toMatchSnapshot();
    });

    it('should save user\'s name to state', () => {
      const wrapper = shallow(<Landing />);
      const expected = 'Han';
      const mockEvent = { preventDefault() { }, target: {name: 'userName', value: 'Han'} };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('userName')).toEqual(expected);
    })

    it('should update the user\'s favorite quote in state', () => {
      const wrapper = shallow(<Landing />);
      const mockEvent = { preventDefault() { }, target: {name: 'userQuote', value: 'Luke, I am your father.'} };
      const expected = 'Luke, I am your father.';

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('userQuote')).toEqual(expected);
    })

    it('should update the user\'s skill level in state', () => {
      const wrapper = (<Landing />);
      const mockEvent = { preventDefault() { }, target: {name: 'skillLevel', value: 'Padawan'} };
      const expected = 'Padawan';

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('skillLevel')).toEqual(expected);
    })
})