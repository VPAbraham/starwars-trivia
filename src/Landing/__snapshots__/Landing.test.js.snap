// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
<section
  className="section section--landing-form"
>
  <img
    alt="star wars"
    className="img img--logo-main"
    src="StarWarsLogo.png"
  />
  <h2>
    Nerd Trove
  </h2>
  <form>
    <label
      htmlFor="userName"
    >
      Enter your name Jedi
    </label>
    <input
      name="userName"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="movieQuote"
    >
      Enter your favorite Star Wars quote
    </label>
    <input
      name="userQuote"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="skillLevel"
    >
      What is your strength in the force?
    </label>
    <select
      name="skillLevel"
      onChange={[Function]}
      required={true}
    >
      <option
        value=""
      >
        Select your strength in the force...
      </option>
      <option
        value="padawan"
      >
        Padawan
      </option>
      <option
        value="jedi knight"
      >
        Jedi Knight
      </option>
      <option
        value="jedi master"
      >
        Jedi Master
      </option>
    </select>
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </form>
  <footer
    className="footer footer--user-bar"
  >
    <p
      className="p p--user-info"
    >
      User Information Bar
    </p>
  </footer>
</section>
`;
